src/Marten/Internal/CompiledQueries/NewCode.cs

```diff
@@ -0,0 +1,437 @@
+            {
+                var queryMember = typeof(PropertyQueryMember<>).CloseAndBuildAs<IQueryMember>(member, memberType);
+                QueryMembers.Add(queryMember);
+            }
+            else if (member is FieldInfo)
+            {
+                var queryMember = typeof(FieldQueryMember<>).CloseAndBuildAs<IQueryMember>(member, memberType);
+                QueryMembers.Add(queryMember);
+            }
+        }
+    }
+
+    private IEnumerable<MemberInfo> findMembers()
+    {
+        foreach (var field in QueryType.GetFields(BindingFlags.Instance | BindingFlags.Public)
+                     .Where(x => !x.HasAttribute<MartenIgnoreAttribute>())) yield return field;
+
+        foreach (var property in QueryType.GetProperties(BindingFlags.Instance | BindingFlags.Public)
+                     .Where(x => !x.HasAttribute<MartenIgnoreAttribute>())) yield return property;
+    }
+
+    #endregion
+
+    public string TenantId { get; set; }
+
+    #region ICommandBuilder implementation
+
+    private CommandPlan _current;
+
+    string ICommandBuilder.LastParameterName => _current.Parameters.LastOrDefault()?.Parameter.ParameterName;
+
+    private CommandPlan appendCommand()
+    {
+        var plan = new CommandPlan();
+        _commands.Add(plan);
+
+        return plan;
+    }
+
+    void ICommandBuilder.Append(string sql)
+    {
+        _current ??= appendCommand();
+
+        _current.CommandText += sql;
+    }
+
+    void ICommandBuilder.Append(char character)
+    {
+        _current ??= appendCommand();
+        _current.CommandText += character;
+    }
+
+    private int _parameterIndex = 0;
+
+    NpgsqlParameter ICommandBuilder.AppendParameter(object value)
+    {
+        _current ??= appendCommand();
+        var name = "p" + _parameterIndex;
+        _parameterIndex++;
+        var usage = new ParameterUsage(_current.Parameters.Count, name, value);
+        _current.Parameters.Add(usage);
+
+        _current.CommandText += ParameterPlaceholder;
+
+        return usage.Parameter;
+    }
+
+    NpgsqlParameter ICommandBuilder.AppendParameter(object value, NpgsqlDbType? dbType)
+    {
+        return appendParameter(value, dbType);
+    }
+
+    private NpgsqlParameter appendParameter(object value, NpgsqlDbType? dbType)
+    {
+        _current ??= appendCommand();
+        var name = "p" + _parameterIndex;
+        _parameterIndex++;
+        var usage = new ParameterUsage(_current.Parameters.Count, name, value, dbType);
+        _current.Parameters.Add(usage);
+
+        _current.CommandText += ParameterPlaceholder;
+
+        return usage.Parameter;
+    }
+
+    void ICommandBuilder.AppendParameters(params object[] parameters)
+    {
+        _current ??= appendCommand();
+        throw new NotSupportedException();
+    }
+
+    NpgsqlParameter[] ICommandBuilder.AppendWithParameters(string text)
+    {
+        _current ??= appendCommand();
+        var split = text.Split('?');
```

Comment:
This is to honor existing customer code?

Reply:
It's used quite a bit in the generated code